src/foam/nanos/controller/ApplicationController.js

```diff
@@ -584,8 +584,10 @@ foam.CLASS({
       for ( var i = 0 ; i < this.MACROS.length ; i++ ) {
         let m = this.MACROS[i];
         text2 = this.expandShortFormMacro(this.expandLongFormMacro(text, m), m);
+        if ( text != text2 )
```

Comment:
Is this 'if' statement necessary?

Reply:
Yeah, this is what was originally causing bugs. Since there could be multiple macro replacements in the string